libpimeval/src/pimPerfEnergyBitSerial.cpp

```diff
@@ -141,30 +142,68 @@ pimPerfEnergyBitSerial::getPerfEnergyForRedSum(PimCmdEnum cmdType, const pimObjI
-        // For a single row, popcount is calculated per 64-bit chunks, and result is shifted then added to an 64-bit accumulator register
-        // If there are multiple regions per core, the multi-region reduction sum is stored in the accumulator
-        double mjEnergyPerPcl = m_pclNsDelay * m_pclUwPower * 1e-12;
-        int numPclPerCore = (maxElementsPerRegion + 63) / 64; // number of 64-bit popcount needed for a row
-        msRuntime = m_tR + (m_pclNsDelay * 1e-6) * numPclPerCore;
-        msRuntime *= bitsPerElement * numPass;
-        mjEnergy = m_eAP * numCore + mjEnergyPerPcl * numPclPerCore * numCore; // energy of one row read and row-wide popcount
-        mjEnergy *= bitsPerElement * numPass;
-        // reduction for all regions
-        double aggregateMs = static_cast<double>(numCore) / 3200000;
-        msRuntime += aggregateMs;
-        mjEnergy += aggregateMs * cpuTDP;
-        mjEnergy += m_pBChip * m_numChipsPerRank * m_numRanks * msRuntime;
-      } else if (dataType == PIM_FP32 || dataType == PIM_FP16 || dataType == PIM_BF16 || dataType == PIM_FP8) {
-        std::cout << "PIM-Warning: Perf energy model for FP reduction sum on bit-serial PIM is not available yet." << std::endl;
-        msRuntime = 999999999.9; // todo
-        mjEnergy = 999999999.9; // todo
+        switch (cmdType)
+        {
+        case PimCmdEnum::REDSUM:
+        case PimCmdEnum::REDSUM_RANGE:
+        {
+          // Assume row-wide popcount capability for integer reduction, with a 64-bit popcount logic unit per PIM core
+          // For a single row, popcount is calculated per 64-bit chunks, and result is shifted then added to an 64-bit accumulator register
+          // If there are multiple regions per core, the multi-region reduction sum is stored in the accumulator
+          double mjEnergyPerPcl = m_pclNsDelay * m_pclUwPower * 1e-12;
+          int numPclPerCore = (maxElementsPerRegion + 63) / 64; // number of 64-bit popcount needed for a row
+          msRuntime = m_tR + (m_pclNsDelay * 1e-6) * numPclPerCore;
+          msRuntime *= bitsPerElement * numPass;
+          mjEnergy = m_eAP * numCore + mjEnergyPerPcl * numPclPerCore * numCore; // energy of one row read and row-wide popcount
+          mjEnergy *= bitsPerElement * numPass;
+          // reduction for all regions
+          double aggregateMs = static_cast<double>(numCore) / 3200000;
+          msRuntime += aggregateMs;
+          mjEnergy += aggregateMs * cpuTDP;
+          mjEnergy += m_pBChip * m_numChipsPerRank * m_numRanks * msRuntime;
+          break;
+        }
+        case PimCmdEnum::REDMIN:
+        case PimCmdEnum::REDMIN_RANGE:
+        {
+          // Reduction tree approach for min/max
+          unsigned levels = static_cast<unsigned>(std::ceil(std::log2(numElements))); // Tree depth
+          pimeval::perfEnergy perfEnergyBS = getPerfEnergyBitSerial(m_simTarget, PimCmdEnum::MIN, dataType, bitsPerElement, numPass, obj);
```

Comment:
We need to add an entry for reduction Min/Max in the bitserial table and then `cmdType` should be passed.